contracts/tokenomics/maker/src/contract.rs

```diff
@@ -352,10 +361,84 @@ fn swap(
     cfg: &Config,
     from_token: AssetInfo,
     amount_in: Uint128,
+) -> Result<SwapTarget, ContractError> {
+    // 1. check direct pair with ASTRO
+    let astro = token_asset_info(cfg.astro_token_contract.clone());
+
+    let direct_pool = query_pair_info(
+        &deps.querier,
+        cfg.factory_contract.clone(),
+        &[from_token.clone(), astro.clone()],
+    );
+
+    if direct_pool.is_ok() {
+        let msg = build_swap_msg(deps, cfg, direct_pool.unwrap(), from_token, amount_in)?;
+        return Ok(SwapTarget::Astro(msg));
+    }
+
+    let uusd = native_asset_info(UUSD_DENOM.to_string());
+    let uluna = native_asset_info(ULUNA_DENOM.to_string());
+
+    // 2. If not ust, try to ust
+    let uusd_pool = uusd_pool_exists(deps, cfg, from_token.clone());
+    if let Some(uusd_pool) = uusd_pool {
+        let msg = build_swap_msg(deps, cfg, uusd_pool, from_token, amount_in)?;
+
+        return Ok(SwapTarget::Bridge { asset: uusd, msg });
+    }
+
+    // 3. try to luna
+    let uluna_pool = uluna_uusd_pools_exist(deps, cfg, from_token.clone());
+    if let Some(uluna_pool) = uluna_pool {
+        let msg = build_swap_msg(deps, cfg, uluna_pool, from_token, amount_in)?;
+
+        return Ok(SwapTarget::Bridge { asset: uluna, msg });
+    }
+
+    // 4. try via bridges
+    let bridge_token = BRIDGES.load(deps.storage, from_token.to_string());
+
+    if bridge_token.is_ok() {
+        let bridge_token = bridge_token.unwrap();
+        let bridge_pool = validate_bridge(
+            deps,
+            cfg,
+            from_token.clone(),
+            bridge_token.clone(),
+            astro.clone(),
+            BRIDGES_INITIAL_DEPTH,
+        )?;
+
+        let msg = build_swap_msg(deps, cfg, bridge_pool, from_token, amount_in)?;
+        return Ok(SwapTarget::Bridge {
```

Comment:
Lets compose this line and line 419 into if else branches and remove `return ... ;` in both of them.